tests/js/chartAuthoringTests.js

```diff
@@ -155,8 +159,50 @@ https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.tx
         jqUnit.assertDeepEq("The reset has an aria-controls attribute properly referencing the form, pie, legend and total", resetAriaControlsAttr, dataEntryFormlId + " " + legendTableId + " " + pieChartPieId + " " + dataEntryFormTotalId);
     };
 
+    // Verify that the updateDataEntryPanel function updates the UI
+    floe.tests.chartAuthoringTester.verifyUpdateDataEntryPanel = function (that) {
+        that.updateDataEntryPanelFromDataSet(floe.tests.chartAuthoring.updateDataSet);
+
+        var dataEntries = that.chartAuthoringInterface.dataEntryPanel.locate("dataEntry");
+
+        var dataEntryLabelSelector = that.chartAuthoringInterface.dataEntryPanel.dataEntry.options.selectors.label;
+        var dataEntryValueSelector = that.chartAuthoringInterface.dataEntryPanel.dataEntry.options.selectors.value;
+
+        dataEntries.each(function (idx) {
+
+            // Confirm that each entry in updateDataSet has a matching dataEntry
+            // in the panel UI, or that the "matching" dataEntry is blank when
+            // the dataset is shorter than the number of possible data entries
+
+            var expectedData = idx < floe.tests.chartAuthoring.updateDataSet.length ? floe.tests.chartAuthoring.updateDataSet[idx] : expectedData = {label: "", value: ""};
+
+            jqUnit.assertEquals("Testing updated dataEntryPanel UI - displayed label at position " + idx + " matches updated data set label", $(this).find(dataEntryLabelSelector).val(), String(expectedData.label));
```

Comment:
At line 179, 180, do you think it's possible to replace ```$(this).find``` with ```that.locate()```? The latter ensures the selector lookup is within ```that.container```, meaning it would exclude tags on the same page with the same selectors but not inside the scope of ```that.container```.